docs/data/date-pickers/digital-clock/digital-clock.md

```diff
@@ -0,0 +1,61 @@
+---
+product: date-pickers
+title: React Digital Clock component
+components: DigitalClock, MultiSectionDigitalClock
+githubLabel: 'component: TimePicker'
+packageName: '@mui/x-date-pickers'
+---
+
+# Digital Clock
+
+<p class="description">The Digital Clock components let the user select a time without any input or popper / modal.</p>
+
+## Description
+
+There are two component versions for different cases. The `DigitalClock` handles selection of a single time instance in one step, just like a `select` component. The `MultiSectionDigitalClock` allows selecting time using separate sections for separate views.
+
+The `DigitalClock` is more appropriate when there is a limited amount of time options needed, while the `MultiSectionDigitalClock` is suited for cases when a more granular time selection is needed.
+
+## Basic usage
+
+{{"demo": "DigitalClockBasic.js"}}
+
+## Uncontrolled vs. Controlled
+
+The components can be uncontrolled or controlled.
+
+{{"demo": "DigitalClockValue.js"}}
+
+## Form props
+
+The components can be disabled or read-only.
+
+{{"demo": "DigitalClockFormProps.js"}}
+
+## Views
+
+The `MultiSectionDigitalClock` component can contain three views: `hours`, `minutes`, and `seconds`.
+By default, only the `hours` and `minutes` views are enabled.
+
+You can customize the enabled views using the `views` prop.
+Views will appear in the order they're included in the `views` array.
+
+{{"demo": "DigitalClockViews.js"}}
+
+## 12h/24h format
+
+By default, the components use the hour format of the locale's time setting, i.e. the 12-hour or 24-hour format.
+
+You can force a specific format using the `ampm` prop.
+
+You can find more information about 12h/24h format in the [Date localization page](/x/react-date-pickers/adapters-locale/#12h-24h-format).
+
+{{"demo": "DigitalClockAmPm.js"}}
+
+## Time step
+
+By default, the components list the time options in `5` unit (`minutes` or `seconds`) intervals.
+
+You can force a desired interval using the `timeStep` prop.
+
+{{"demo": "DigitalClockTimeStep.js"}}
```

Comment:
I'd expect that a value of 120 would also skip hours in the `MultiSectionDigitalClock`. Am I missing something?
https://codesandbox.io/s/timestep120-hyocvd

Also, what If users need only even hours (7, 9, 11)? Is there a way to achieve that?
For example, a conference-room booking system that has 2 hours slots, from 9:00 to 17:00.

Reply:
Hmmm, that's a good observation.
I do not think it would currently be possible
Maybe we'd need the option to provide an additional prop controlling the start/offset of the first-hour option? 🤔 
Or, maybe we could add a property `skipDisabled` or something along those lines and achieve it by mixing `min/maxTime` together with this prop. 🤔 
What do you think @flaviendelangle @alexfauquette?

Reply:
I'm also a bit disturbed by the `timeStep` being applied to both minutes and seconds. If your precision is more than 1 minute, I don't see the point to provide the number of seconds. This time step should probably set only to the smallest amount of time.

My first guest is that it could get two sets of options:
- `hoursStep`, `minutesStep`, `secondsStep` (or an object `timeStep={{hours: 1, minutes: 15 }}`)
- `hoursOptions`, `minutesOptions`, `secondsOptions` an array of values to display

Reply:
How do you want to handle it for the mobile view ?
Maybe we could support both format on `TimeClock` and unify on v7

Reply:
> How do you want to handle it for the mobile view ? Maybe we could support both format on `TimeClock` and unify on v7

I was planning to not support it in `TimeClock` at all. What do you think? 🤔

Reply:
Don't we have the same topic on the `TimeClock` where people currently can't differentiate the step across views ?

Not saying that it must be done in this PR of course

Reply:
Indeed, we can think of extending TimeClock functionality if we see the need for it later on.
But at its current state, I'm excluding props of the digital clock from `Mobile` picker versions. 🤔

Reply:
OK
Maybe it's worth having an explicit warning on both JSDoc to help people understand why the props are not named the same way and don't behave the same way.

Because most people are using the responsive version and will be surprised if they have something working on desktop and start having invalid data sent to there server because the mobile version does not behave the same way.

Reply:
> Maybe we'd need the option to provide an additional prop controlling the start/offset of the first-hour option? 🤔
Or, maybe we could add a property skipDisabled or something along those lines and achieve it by mixing min/maxTime together with this prop. 

Interesting, mixing min/maxTime and using min as the start point seems intuitive enough, but isn't the timeStep already "skipping the disabled", for example in case you set 120 minutes on the `DigitalClock`? 
Would you show the options in between if it's set to false?

> `timeStep={{hours: 1, minutes: 15 }})`

That seems flexible. But the behavior implemented is a bit odd (also on kendo UI), for example when `timeStep={{minutes: 45}})` the selection box displays 00 and 45 minutes on the field, but I'd expect that some hours need to be set with 30 minutes, like 12:00 -> 12:45 -> 13:30
to illustrate the example: https://codesandbox.io/s/timestep45-sbuqsp

Reply:
@joserodolfofreitas I'm currently treating the `timeSteps` for each of the sections as separate settings.
Basically, each field setting affects only the relevant section time stepping.
Do you think it should work differently? If yes, maybe this behavior should be controllable? 🤔

Reply:
What do you think @joserodolfofreitas, does having a different type and name (timeStep: number) for `DigitalClock` is worth the effort? 🤔 
https://github.com/mui/mui-x/pull/7958/commits/72da1ff1d0640c859b65f0260a5650464cb2e832

Reply:
> I wish we could somehow warn the user they are using the wrong property. Based on type only, the IDE will only say that DigitalClock doesn't have timeSteps property and vice versa.

I don't have the biggest fear about these props names as they are only different on the separate clock components, which should rarely be used standalone. When using the `TimePicker` the API will be the same and the fact that [the `DigitalClock` will only use `timeSteps.minutes`](https://deploy-preview-7958--material-ui-x.netlify.app/x/api/date-pickers/time-picker/) is denoted in the JSDoc comment.